### PR TITLE
json-fortran-8.3.0-GCC-10.3.0.eb

### PR DESCRIPTION
**Pull request by moravveji**
This PR includes `json-fortran-8.3.0-GCCcore-10.3.0.eb`, which is fully based on an existing easyconfig file for the module `json-fortran/8.3.0-GCCcore-11.2.0`. Only the choice of toolchain year and dependency versions are adapted to the 2021a toolchain.

### CODE REVIEW

**Comment by ocaisa** [comment]
This should be targeted at the GCC toolchain as it produces fortran modules (so cannot be used as a dependency for anything with an intel compiler)

**Comment by ocaisa** [comment]
I've updated the easyconfigs upstream to reflect this: https://github.com/easybuilders/easybuild-easyconfigs/pull/19283

**Comment by moravveji** [comment]
> I've updated the easyconfigs upstream to reflect this: #19283

@ocaisa : thanks for your comments on this PR. The commit `dcd8ed8ecc` shall address your requested change.
Don't you mind to also review this PR?

File: easybuild/easyconfigs/j/json-fortran/json-fortran-8.3.0-GCC-10.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+toolchain = {'name': 'GCC', 'version': '10.3.0'}
+
+source_urls = ['https://github.com/jacobwilliams/json-fortran/archive/']
+sources = ['%(version)s.tar.gz']
+checksums = ['5fe9ad709a726416cec986886503e0526419742e288c4e43f63c1c22026d1e8a']
+
+builddependencies = [
+    ('binutils', '2.36.1', '', ('GCCcore', '10.3.0')),
+    ('CMake', '3.20.1', '', ('GCCcore', '10.3.0')),
```
Comment (ocaisa) [inline easybuild/easyconfigs/j/json-fortran/json-fortran-8.3.0-GCC-10.3.0.eb:19]:
```suggestion
    ('binutils', '2.36.1'),
    ('CMake', '3.20.1'),
```
These dependencies will get picked up automatically (`GCCcore` is a subtoochain of `GCC`)

**Review by ocaisa — CHANGES_REQUESTED** [review]

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Review by ocaisa — APPROVED** [review]

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2

**Comment by ocaisa** [comment]
Going in, thanks @moravveji

**Pull request merged**